### PR TITLE
Add payment processor list endpoint.

### PR DESCRIPTION
**Pull request by jimabramson**
XCOM-211

@clintonb @rlucioni please review.

### CODE REVIEW

File: ecommerce/extensions/payment/tests/test_views.py [hunk 1]
```diff
@@ -81,6 +84,32 @@ def handle_processor_response(self, params):
...
+        """
+        ProcessorsListView._names = None
+
+    @override_settings(PAYMENT_PROCESSORS=[SUCCESS_PROCESSOR])
+    def test_get_one(self):
+        """
+        Ensure a single payment processor in settings is handled correctly.
+        """
+        response = self.client.get(reverse('processors_list'))
+        self.assertEqual(json.loads(response.content), ['DummySuccessProcessor'])
```
Comment (clintonb) [inline ecommerce/extensions/payment/tests/test_views.py]: [Nice to have] DRY: Combine the get and assertion into a single helper method that accepts the expected list as input:

``` python
def assert_processor_list_matches(expected):
  ...
```

File: ecommerce/extensions/payment/views.py [hunk 2]
```diff
@@ -12,6 +12,25 @@
 from ecommerce.extensions.payment.helpers import get_processor_class
 
 
+class ProcessorsListView(View):
```
Comment (clintonb) [inline ecommerce/extensions/payment/views.py]: We use Django REST Framework for our API views. (Disregard the one below, which I will replace.) In this instance a `[ListAPIView](http://www.django-rest-framework.org/api-guide/generic-views/#listapiview)` is ideal.

You'll need to do the following:
1. Override `get_queryset` to return the list of processor classes.
2. Add a [serializer](http://www.django-rest-framework.org/api-guide/serializers/) (e.g. PaymentProcessorSerializer) that exposes the name field.

This is a lot of code; but, we get the included authentication bits that we already use. (See the order API tests for examples of testing this authentication.) Also, let me know if you already know DRF and I just listed all of this for nothing. :/

Comment (jimabramson) [inline ecommerce/extensions/payment/views.py]: @clintonb no, thanks, appreciate the tips.  initially figured this data was trivial enough to sidestep some formalities, but i do see the benefit in doing things consistently.

**Comment by jimabramson** [comment]
ready for re-review.

File: ecommerce/extensions/payment/tests/test_views.py [hunk 3]
```diff
@@ -81,6 +85,36 @@ def handle_processor_response(self, params):
...
+
+    def setUp(self):
+        self.token = self.generate_jwt_token_header(self.create_user())
+
+    def assert_processor_list_matches(self, expected):
+        """ DRY helper. """
+        response = self.client.get(reverse('processor_list'), HTTP_AUTHORIZATION=self.token)
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.content)
+        self.assertEqual(data, {'count': len(expected), 'previous': None, 'next': None, 'results': ANY})
```
Comment (clintonb) [inline ecommerce/extensions/payment/tests/test_views.py]: Doh (on me)! We have pagination setup by default, but I would be okay disabling it for this endpoint. I can't imagine we'd ever have so many payment processors that we need to paginate them. I believe if you set the `pagination_class` on your view class to `None`, the view should return a list as you originally tested. See http://www.django-rest-framework.org/api-guide/pagination/ for additional info.

**Comment by clintonb** [comment]
:+1: once pagination removed. (Also, why are we all working on Saturday!? My excuse is that I'm also watching Netflix.)

**Comment by rlucioni** [comment]
:+1: My excuse is that I'm waiting for laundry to finish drying.

File: ecommerce/extensions/payment/tests/test_views.py [hunk 4]
```diff
@@ -81,6 +85,34 @@ def handle_processor_response(self, params):
...
+    """ Ensures correct behavior of the payment processors list view."""
+
+    def setUp(self):
+        self.token = self.generate_jwt_token_header(self.create_user())
+
+    def assert_processor_list_matches(self, expected):
+        """ DRY helper. """
+        response = self.client.get(reverse('processor_list'), HTTP_AUTHORIZATION=self.token)
+        self.assertEqual(response.status_code, 200)
+        self.assertSetEqual(set(json.loads(response.content)), set(expected))
```
Comment (rlucioni) [inline ecommerce/extensions/payment/tests/test_views.py:97]: Minor: It shouldn't be necessary to invoke `assertSetEqual` directly. `assertEqual` knowns to use the right type-specific equality method ([reference](https://docs.python.org/2/library/unittest.html#unittest.TestCase.addTypeEqualityFunc)).

**Comment by clintonb** [comment]
@rlucioni I'm doing laundry too...and it's been sitting in the washer for a couple hours instead of drying. Oops.

**Comment by jimabramson** [comment]
wait...it's Saturday?   ugh.

**Comment by clintonb** [comment]
Not sure if he's joking. :confused:

**Comment by jimabramson** [comment]
joking :-)

**Pull request merged**